pkg/controller/datavolume/util.go

```diff
@@ -297,12 +304,70 @@ func resolveVolumeSize(c client.Client, dvSpec cdiv1.DataVolumeSpec, pvcSpec *v1
 		return nil, errors.Errorf("Datavolume Spec is not valid - missing storage size")
 	}
 
+	if !noInflate {
+		return &requestedSize, nil
+	}
 	// disk or image size, inflate it with overhead
 	requestedSize, err := cc.InflateSizeWithOverhead(c, requestedSize.Value(), pvcSpec)
 
 	return &requestedSize, err
 }
 
+// storageClassCSIDriverExists returns true if the passed storage class has CSI drivers available
+func storageClassCSIDriverExists(client client.Client, log logr.Logger, storageClassName *string) (bool, error) {
+	log = log.WithName("getCsiDriverForStorageClass").V(3)
+
+	storageClass, err := cc.GetStorageClassByName(client, storageClassName)
+	if err != nil {
+		return false, err
+	}
+	if storageClass == nil {
+		log.Info("Target PVC's Storage Class not found")
+		return false, nil
+	}
+
+	csiDriver := &storagev1.CSIDriver{}
+
+	if err := client.Get(context.TODO(), types.NamespacedName{Name: storageClass.Provisioner}, csiDriver); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return false, err
+		}
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func checkPVCUsingPopulators(pvc *v1.PersistentVolumeClaim) (bool, error) {
+	if pvc.Spec.DataSourceRef == nil {
+		return false, nil
+	}
+	usePopulator, ok := pvc.Annotations[cc.AnnUsePopulator]
+	if !ok {
+		return false, nil
+	}
+	boolUsePopulator, err := strconv.ParseBool(usePopulator)
+	if err != nil {
+		return false, err
+	}
+	return boolUsePopulator, nil
+}
+
+func shouldUseCDIPopulator(client client.Client, log logr.Logger, dv *cdiv1.DataVolume, storageClass *string) (bool, error) {
+	if usePopulator, ok := dv.Annotations[cc.AnnUsePopulator]; ok {
+		boolUsePopulator, err := strconv.ParseBool(usePopulator)
+		if err != nil {
+			return false, err
+		}
+		return boolUsePopulator, nil
+	}
+	usePopulator, err := storageClassCSIDriverExists(client, log, storageClass)
+	if err != nil {
+		return false, err
+	}
+	cc.AddAnnotation(dv, cc.AnnUsePopulator, strconv.FormatBool(usePopulator))
```

Comment:
I don't love that this function has the side effect of mutating the datavolume, perhaps it would be better to explicitly set this somehow in one place?  Currently, this function is called in multiple places

Reply:
I guess I can do the update itself on a line after the call to this line and only in one place in the syncDvPvcState function after updating syncState.usePopulator